pkg/mcp/server/server.go

```diff
@@ -137,12 +169,16 @@ func (s *Server) newConnection(stream mcp.AggregatedMeshConfigService_StreamAggr
 	}
 
 	scope.Infof("MCP: connection %v: NEW, supported types: %#v", con, messageNames)
-	return con
+	return con, nil
 }
 
 // StreamAggregatedResources implements bidirectional streaming method for MCP (ADS).
 func (s *Server) StreamAggregatedResources(stream mcp.AggregatedMeshConfigService_StreamAggregatedResourcesServer) error { // nolint: lll
-	con := s.newConnection(stream)
+	con, err := s.newConnection(stream)
+	if err != nil {
```

Comment:
We should probably return a proper gRPC status code that indicates authentication failed, e.g. `UNAUTHENTICATED`.